src/main/java/com/viaversion/viafabricplus/injection/mixin/features/block/shape/MixinDoorBlock.java

```diff
@@ -0,0 +1,77 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.viaversion.viafabricplus.injection.mixin.features.block.shape;
+
+import com.viaversion.viafabricplus.protocoltranslator.ProtocolTranslator;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.DoorBlock;
+import net.minecraft.block.ShapeContext;
+import net.minecraft.block.enums.DoorHinge;
+import net.minecraft.state.property.BooleanProperty;
+import net.minecraft.state.property.EnumProperty;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.shape.VoxelShape;
+import net.minecraft.util.shape.VoxelShapes;
+import net.minecraft.world.BlockView;
+import net.raphimc.viabedrock.api.BedrockProtocolVersion;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import java.util.Map;
+
+@Mixin(DoorBlock.class)
+public abstract class MixinDoorBlock {
+
+    @Unique
+    private static final Map<Direction, VoxelShape> viaFabricPlus$shape_bedrock = Map.of(
+        Direction.NORTH, VoxelShapes.cuboid(0, 0, 0, 1, 1, 0.1825F),
+        Direction.SOUTH, VoxelShapes.cuboid(0, 0, 0.8175F, 1, 1, 1),
+        Direction.WEST, VoxelShapes.cuboid(0.8175F, 0, 0, 1, 1, 1),
+        Direction.EAST, VoxelShapes.cuboid(0, 0, 0, 0.1825F, 1, 1)
+    );
+
+    @Shadow
+    @Final
+    public static EnumProperty<Direction> FACING;
+
+    @Shadow
+    @Final
+    public static BooleanProperty OPEN;
+
+    @Shadow
+    @Final
+    public static EnumProperty<DoorHinge> HINGE;
+
+    @Inject(method = "getOutlineShape", at = @At(value = "RETURN"), cancellable = true)
+    private void modifyCollisionShape(BlockState state, BlockView world, BlockPos pos, ShapeContext context, CallbackInfoReturnable<VoxelShape> cir) {
+        if (ProtocolTranslator.getTargetVersion().equals(BedrockProtocolVersion.bedrockLatest)) {
+            Direction direction = state.get(FACING);
+            Direction direction2 = (Boolean)state.get(OPEN) ? (state.get(HINGE) == DoorHinge.RIGHT ? direction.rotateYCounterclockwise() : direction.rotateYClockwise()) : direction;
+            cir.setReturnValue(viaFabricPlus$shape_bedrock.get(direction2));
```

Comment:
Maybe Redirect on FIELD and just replace the shape if the Direction snippet is copied from the class